jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -257,15 +257,13 @@ def _parse_hdulist(app, hdulist, file_name=None,
         if hdu.name != 'PRIMARY' and 'PRIMARY' in hdulist:
             metadata[PRIHDR_KEY] = standardize_metadata(hdulist['PRIMARY'].header)
 
-        sc = _return_spectrum_with_correct_units(flux, wcs, metadata, data_type, hdulist=hdulist)
-
         # store original WCS in metadata. this is a hacky workaround for converting subsets
         # to sky regions, where the parent data of the subset might have dropped spatial WCS info
         metadata['_orig_spatial_wcs'] = _get_celestial_wcs(wcs)
 
+        sc = _return_spectrum_with_correct_units(flux, wcs, metadata, data_type, hdulist=hdulist)
+
         app.add_data(sc, data_label)
-        if data_type == 'flux':  # Forced wave unit conversion made it lose stuff, so re-add
```

Comment:
This one was blamed to me back in #1480 but I have no recollection writing it... It was @javerbukh's PR so maybe he remembers why we did it this way or whether we can move it further down. 🤷‍♀️